logging/impl.go

```diff
@@ -35,34 +35,47 @@ type (
 	}
 )
 
-func (impl *impl) NewLogEntry() *LogEntry {
+func (imp *impl) NewLogEntry() *LogEntry {
 	ret := &LogEntry{}
 	ret.Time = time.Now()
-	ret.LoggerName = impl.name
+	ret.LoggerName = imp.name
 	ret.Caller = getCaller()
 
 	return ret
 }
 
-func (impl *impl) AddAppender(appender Appender) {
-	impl.appenders = append(impl.appenders, appender)
+func (imp *impl) AddAppender(appender Appender) {
+	imp.appenders = append(imp.appenders, appender)
 }
 
-func (impl *impl) Desugar() *zap.Logger {
-	return nil
+func (imp *impl) Desugar() *zap.Logger {
+	return imp.AsZap().Desugar()
 }
 
-func (impl *impl) Level() zapcore.Level {
-	return zapcore.InfoLevel
+func (imp *impl) Level() zapcore.Level {
+	return imp.level.AsZap()
 }
 
-func (impl *impl) Named(name string) *zap.SugaredLogger {
-	return nil
+func (imp *impl) Sublogger(subname string) Logger {
```

Comment:
`Sublogger` is just a different name for `Named`, but it returns a `Logger` as opposed to a `zap.SugaredLogger`.

After removing `ZapCompatibleLogger`, we're free to go back to `Named` if we prefer that. But for this (and potentially some other things coming up trying to replace zap) we may need to "duplicate" functions with just their return value substituted.